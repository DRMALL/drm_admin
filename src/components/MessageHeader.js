

import React from 'react'
import { Link } from 'react-router'
import dispatch from '../actions/dispatch'

const MessageHeader = ({ state }) => (
  <div className='message-header-container' >
    <div className='message-header-left' >
      <div className={ state.message.messageSelect==='notsend' ? 'message-header-item message-header-select' : 'message-header-item message-header-noselect' }
           onClick={ messageSelect } data-id='未发送' >未发送</div>
      <div className={ state.message.messageSelect==='send' ? 'message-header-item message-header-select' : 'message-header-item message-header-noselect' }
           onClick={ messageSelect } data-id='已发送' >已发送</div>
    </div>
    <Link to='/message/new' className='message-header-right'  >
      <div className='message-header-icon' >
        <i className="iconfont icon-tianjiajiahaowubiankuang"  ></i>
      </div>
      <div className='message-header-text'  >创建信息</div>
    </Link>
  </div>
  )

export default MessageHeader

function messageSelect(e){
  const { id } = e.currentTarget.dataset
  if(id==='未发送'){
    dispatch('MESSAGE_SELECT_SEND_FASLE')
  }
  else
  dispatch('MESSAGE_SELECT_SEND_TRUE')
}
