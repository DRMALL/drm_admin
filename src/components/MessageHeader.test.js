

import React from 'react'
import MessageHeader from './MessageHeader'
import renderer from 'react-test-renderer'

it('MessageHeader test', () => {
  const state = { message: { messageSelect: 'notsend' } }
  const tree = renderer.create(<MessageHeader state={ state } /> )
  expect(tree).toMatchSnapshot()
})
