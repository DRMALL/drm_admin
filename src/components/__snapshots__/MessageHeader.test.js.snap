// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageHeader test 1`] = `
<div
  className="message-header-container"
>
  <div
    className="message-header-left"
  >
    <div
      className="message-header-item message-header-select"
      data-id="未发布"
      onClick={[Function]}
    >
      未发布
    </div>
    <div
      className="message-header-item message-header-noselect"
      data-id="已发布"
      onClick={[Function]}
    >
      已发布
    </div>
  </div>
  <div
    className="message-header-link"
  >
    <a
      className="content-link"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="content-link-block"
      >
        <i
          className="anticon anticon-plus"
        />
      </div>
      <div
        className="content-link-text"
      >
        创建信息
      </div>
    </a>
  </div>
</div>
`;
